elogia_hr/models/hr.py

```diff
@@ -40,18 +52,352 @@ class EmployeeHistory(models.Model):
-                                              ('type_action', '=', self.type_action),
-                                              ('effective_date', '=', self.effective_date)])
-            if obj_history:
-                for item in obj_history:
-                    item.unlink()
+
+class ListActionChange(models.Model):
+    _name = 'list.action.change'
+    _description = 'List Action Change'
+    _rec_name = 'planning_id'
+
+    planning_id = fields.Many2one('employee.change.planning', 'Employee Change', ondelete='cascade')
+    employee_id = fields.Many2one('hr.employee', related='planning_id.employee_id', string='Employee')
+    effective_date = fields.Date('Effective date', states={'not_processed': [('readonly', False)],
+                                                           'processed': [('readonly', True)]})
+    last_date = fields.Date('Last date')
+    type_model = fields.Selection([
+        ('employee', 'Employee'),
+        ('contract', 'Contract'),
+    ], string='Model', default='employee', required=True,
+        help="Type model to update in Contract or Employee History", states={'not_processed': [('readonly', False)],
+                                                                             'processed': [('readonly', True)]})
+    state = fields.Selection([
+        ('not_processed', 'Not processed'),
+        ('processed', 'Processed'),
+    ], string='Status', default='not_processed', required=True)
+    type_action = fields.Selection([
+        ('employee', '-'),
+        ('responsible', 'Responsible'),
+        ('company', 'Company'),
+        ('department', 'Department'),
+        ('rol', 'Job'),
+        ('hub', 'Hub'),
+        ('contract', 'Contratation Type'),
+        ('type_c', 'Type Contract'),
+        ('type_scholarship', 'Scholarship'),
+        ('departure_reason', 'Departure reason'),
+        ('departure_date', 'Departure date'),
+        ('wage_normal', 'Salary'),
+        ('wage_variable', 'Salary V'),
+    ], string='Action', default='employee', required=True,
+        help="Type action to update in Contract or Employee History")
+
+    @api.constrains('type_model')
+    def check_contact(self):
+        for record in self:
+            if record.type_model == 'contract' and not record.employee_id.contract_id:
+                raise UserError(_('The employee %s does not have an active contract.') % record.employee_id.name)
+
+    @api.constrains('state')
+    def onchange_state_planning(self):
+        for record in self:
+            if record.state == 'processed':
+                if not any(item for item in record.planning_id.list_ids if item.state == 'not_processed'):
+                    record.planning_id.state = 'processed'
+
+
+class EmployeeChangePlanning(models.Model):
+    _name = "employee.change.planning"
+    _description = "Employee Change"
+    _rec_name = 'employee_id'
+
+    def _get_company_currency(self):
```

Comment:
funciones para computar campos, puede venir un recordset, entonces debes preveer usando for rec in self. Como nota  para que quede mas organizado este metodo deberia estar abajo y comenzando por _compute, de esta manera es facil identitificar que esta funcion es para computar un campo